src/Microsoft.Azure.EventHubs.Processor/AzureStorageCheckpointLeaseManager.cs

```diff
@@ -241,73 +240,75 @@ public async Task<bool> DeleteLeaseStoreAsync()
 
         public async Task<Lease> GetLeaseAsync(string partitionId) // throws URISyntaxException, IOException, StorageException
         {
-    	    AzureBlobLease retval = null;
+            AzureBlobLease retval = null;
 
             CloudBlockBlob leaseBlob = GetBlockBlobReference(partitionId);
 
             if (await leaseBlob.ExistsAsync(null, this.operationContext).ConfigureAwait(false))
-		    {
+            {
                 retval = await DownloadLeaseAsync(partitionId, leaseBlob).ConfigureAwait(false);
-		    }
+            }
 
             return retval;
         }
 
         public IEnumerable<Task<Lease>> GetAllLeases()
         {
-            List<Task<Lease>> leaseFutures = new List<Task<Lease>>();
-            IEnumerable<string> partitionIds = this.host.PartitionManager.GetPartitionIdsAsync().Result;
+            IEnumerable<string> partitionIds =
+                Task.Run(() => this.host.PartitionManager.GetPartitionIdsAsync())
```

Comment:
>Task.Run [](start = 16, length = 8)

I mean you don't need to use Task.Run. Below does same job.

partitionIds = this.host.PartitionManager.GetPartitionIdsAsync().ConfigureAwait(false).GetAwaiter().GetResult();

Reply:
ohhh Ok